### PR TITLE
Adding account_token_position to account exports

### PR DESCRIPTION
**Pull request by matthiashanel**
Signed-off-by: Matthias Hanel <mh@synadia.com

### CODE REVIEW

File: exports.go [hunk 1]
```diff
@@ -124,6 +126,28 @@ func (e *Export) Validate(vr *ValidationResults) {
 		e.Latency.Validate(vr)
 	}
 	e.Subject.Validate(vr)
+
+	if e.AccountTokenPosition > 0 {
+		if !e.Subject.HasWildCards() {
+			vr.AddError("Account Token Position can only be used when wildcard subjects: %s", e.Subject)
```
Comment (aricart) [inline exports.go:132]: Account token position can onlyh be used 'with' wildcarded subjects?

File: exports.go [hunk 2]
```diff
@@ -124,6 +126,28 @@ func (e *Export) Validate(vr *ValidationResults) {
...
+		} else {
+			subj := string(e.Subject)
+			token := strings.Split(subj, ".")
+			tkCnt := uint(len(token))
+			if e.AccountTokenPosition == tkCnt && strings.HasSuffix(subj, "*") {
```
Comment (aricart) [inline exports.go:137]: I think that the account token position needs to be an independent token and it needs to match `*`. That is if your `account_token_position` is set to `1` and you have a subject like `foo.>`, that should fail validation. The subject in question should be perhaps `foo.*.>` or `foo.*.*`, but at any rate, the intent of what token represents the account should be explicit on the subject template as well.

File: exports.go (hunk 1 above)
Comment (matthiashanel) [reply exports.go:132]: yes, seems pointless without. wouldn't you agree?

File: exports.go (hunk 2 above)
Comment (matthiashanel) [reply exports.go:137]: so you are saying the token position needs to match a * and can't match a token covered by > ?

File: exports_test.go [hunk 3]
```diff
@@ -288,3 +288,71 @@ func TestExport_Sorting(t *testing.T) {
 		t.Fatal("exports not sorted")
 	}
 }
+
+func TestExportAccountTokenPos(t *testing.T) {
+	okp := createOperatorNKey(t)
+	akp := createAccountNKey(t)
+	apk := publicKey(akp, t)
+	tbl := map[Subject]uint{
+		">":           5,
```
Comment (matthiashanel) [inline exports_test.go:297]: @aricart no to this then?

File: exports_test.go [hunk 4]
```diff
@@ -288,3 +288,71 @@ func TestExport_Sorting(t *testing.T) {
...
+		"*":           1,
+		"foo.*":       2,
+		"foo.>":       2,
```
Comment (matthiashanel) [inline exports_test.go:300]: @aricart  no to this as well?

File: exports.go (hunk 2 above)
Comment (matthiashanel) [reply exports.go]: @aricart , made it function with * only

File: exports_test.go (hunk 3 above)
Comment (aricart) [reply exports_test.go]: Yes I don't think so.

File: exports_test.go (hunk 4 above)
Comment (aricart) [reply exports_test.go]: only lines 298 and 299 are valid tbh.

**Pull request merged**